### PR TITLE
Fixed empty list parsing.

### PR DESCRIPTION
**Pull request by burgerkingorama**
See #66.
Sorry for recreating the pull request but the first one was started from the wrong branch.

### CODE REVIEW

File: erpcgen/src/templates/c_coders.template [hunk 1]
```diff
@@ -34,15 +34,17 @@ uint32_t {$info.sizeTemp};
 {% endif %}
 uint8_t * {$info.dataTemp};
 codec->readBinary(&{$info.sizeTemp}, &{$info.dataTemp});
+if({$info.sizeTemp})
```
Comment (MichalPrincNXP) [inline erpcgen/src/templates/c_coders.template:37]: - if statements should have one space char between "if" and "(" 
- the added 2 lines should be moved 3 lines bellow (attaching the expected content)
[c_coders.zip](https://github.com/EmbeddedRPC/erpc/files/5022760/c_coders.zip)

**Comment by Hadatko** [comment]
> kErpcStatus_MemoryError

I would rather update writeData to use if condition inside, not in writeBinaryFile

**Review by Hadatko — APPROVED** [review]
After mine changes i thing it is ready for merge.

**Comment by MichalPrincNXP** [comment]
> After mine changes i thing it is ready for merge.

Dusan, looks good but I would also prefer to have execution tests updates to cover this particular case (not only the erpcgen test). Could you please update them yet (binary and lists I guess)?

**Comment by Hadatko** [comment]
We cannot add test for binary type (we have restriction to use out and inout type in one application for list, binary,... types). But i created test for in and out for list type. We could do probably same for binary but the test would be present in other test scope than test_binary

**Review by Hadatko — APPROVED** [review]

**Review by MichalPrincNXP — APPROVED** [review]

**Comment by MichalPrincNXP** [comment]
Thank you very much!

**Pull request merged**